src/fondant/executor.py

```diff
@@ -424,3 +425,24 @@ def _write_data(self, dataframe: dd.DataFrame, *, manifest: Manifest):
 
     def upload_manifest(self, manifest: Manifest, save_path: t.Union[str, Path]):
         pass
+
+
+COMPONENT_EXECUTOR_MAPPING: t.Dict[str, t.Type[Executor]] = {
+    "DaskLoadComponent": DaskLoadExecutor,
+    "DaskTransformComponent": DaskTransformExecutor,
+    "DaskWriteComponent": DaskWriteExecutor,
+    "PandasTransformComponent": PandasTransformExecutor,
+}
+
+
+class ComponentRunner:
+    def __init__(self, component: t.Type[Component]):
+        self.component = component
+
+    def _get_executor(self) -> Executor:
```

Comment:
If we go for the factory approach, this method would be the only one inside of this class.
Basically, something like this:

```
def create_executer(self, component) -> Executer:
        component_type = self.component.__bases__[0].__name__
        return COMPONENT_EXECUTOR_MAPPING[component_type].from_args()
```